klippy/extras/load_cell_probe.py

```diff
@@ -0,0 +1,377 @@
+        # multi probes state
+        self.multi = 'OFF'
+
+    def _config_commands(self):
+        self._sos_filter.create_filter()
+        sos_filters = self._continuous_tare_filter.get_filter_sections()
+        self._sos_filter.configure_filter(sos_filters)
+        self._mcu.add_config_cmd("config_load_cell_endstop oid=%d"
+                                 " sos_filter_oid=%d"
+                                 % (self._oid, self._sos_filter.oid))
+
+    def _build_config(self):
+        # Lookup commands
+        cmd_queue = self._dispatch.get_command_queue()
+        self._query_cmd = self._mcu.lookup_query_command(
+            "load_cell_endstop_query_state oid=%c",
+            "load_cell_endstop_state oid=%c homing=%c homing_triggered=%c"
+            " is_triggered=%c trigger_ticks=%u sample=%i sample_ticks=%u"
+            , oid=self._oid, cq=cmd_queue)
+        self._set_range_cmd = self._mcu.lookup_command(
+            "set_range_load_cell_endstop"
+            " oid=%c safety_counts_min=%i safety_counts_max=%i tare_counts=%i"
+            " trigger_grams=%u grams_per_count=%i"
+            , cq=cmd_queue)
+        self._home_cmd = self._mcu.lookup_command(
+            "load_cell_endstop_home oid=%c trsync_oid=%c trigger_reason=%c"
+            " error_reason=%c clock=%u rest_ticks=%u timeout=%u"
+            , cq=cmd_queue)
+
+    def _build_sos_filter(self, config):
+        # optional continuous tearing
+        sps = self._load_cell.get_sensor().get_samples_per_second()
+        # Collect 4x60hz power cycles of data to average across power noise
+        default_tare_samples = max(2, round(sps * ((1 / 60) * 4)))
+        self.tare_samples = config.getfloat('tare_samples',
+                                            default=default_tare_samples,
+                                            minval=2, maxval=sps)
+        max_filter_frequency = math.floor(sps / 2.)
+        hp_option = "drift_filter_cutoff_frequency"
+        highpass = config.getfloat(hp_option, minval=0.1, maxval=20.0,
+                                    below=max_filter_frequency, default=None)
+        lowpass = config.getfloat("buzz_filter_cutoff_frequency",
+                                    above=min(80.0, max_filter_frequency - 1.0),
+                                    below=max_filter_frequency, default=None)
+        notches = sos_filter.getfloatlist(config, "notch_filter_frequencies",
+                                    max_len=2,
+                                    above=(highpass or 0.),
+                                    below=(lowpass or max_filter_frequency))
+        notch_quality = config.getfloat("notch_filter_quality",
+                                        minval=0.5, maxval=6.0, default=2.0)
+        return sos_filter.DigitalFilter(sps, config.error, highpass,
+                             lowpass, notches, notch_quality)
+
+    def _handle_ready(self):
+        self._sensor.attach_endstop(self._oid)
```

Comment:
It's not valid to call anything that could fail or delay in the ready callback.  You could do this work from a connect callback, or you could arrange to associate the loadcell with the sensor during the init phase so that the sensor can send the appropriate command to the mcu during its `build_config()` callback.